pxtblocks/blocklyloader.ts

```diff
@@ -540,48 +512,117 @@ namespace pxt.blocks {
+    }
+
     function initLoops() {
+        let msg:any = Blockly.Msg;
+
+        // builtin controls_repeat_ext
+        msg.CONTROLS_REPEAT_TITLE = lf("repeat %1 times");
+        msg.CONTROLS_REPEAT_INPUT_DO = lf("_repeat_do"); // "do" in English
+        installHelpResources(
+            'controls_repeat_ext',
+            lf("a loop that repeats and increments an index"),
+            lf("Do some statements several times."),
+            '/blocks/loops/repeat'
+        );
+
+        // pxt device_while
         Blockly.Blocks['device_while'] = {
             init: function () {
-                this.setHelpUrl('/blocks/loops/while');
-                this.setColour(blockColors['loops']);
-                this.appendValueInput("COND")
-                    .setCheck("Boolean")
-                    .appendField("while");
+                this.jsonInit({
+                  "message0": lf("while %1"),
+                  "args0": [
+                    {
+                      "type": "input_value",
+                      "name": "COND",
+                      "check": "Boolean"
+                    }
+                  ],
+                  "previousStatement": null,
+                  "nextStatement": null,
+                  "colour": blockColors['loops']
+                });
                 this.appendStatementInput("DO")
-                    .appendField("do");
-                this.setPreviousStatement(true);
-                this.setNextStatement(true);
-                this.setTooltip(lf("Run the same sequence of actions while the condition is met."));
+                    .appendField(lf("_while_do")); // "do" in English
+
+                setHelpResources(this,
+                    'device_while',
+                    lf("a loop that repeats while the condition is true"),
+                    lf("Run the same sequence of actions while the condition is met."),
+                    '/blocks/loops/while'
+                );
             }
         };
 
+        // pxt controls_simple_for
         Blockly.Blocks['controls_simple_for'] = {
             /**
              * Block for 'for' loop.
              * @this Blockly.Block
              */
             init: function () {
-                this.setHelpUrl("/blocks/loops/for");
-                this.setColour((<any>Blockly.Blocks).loops.HUE);
-                this.appendDummyInput()
-                    .appendField("for")
-                    .appendField(new Blockly.FieldVariable(null), 'VAR')
-                    .appendField("from 0 to");
-                this.appendValueInput("TO")
-                    .setCheck("Number")
-                    .setAlign(Blockly.ALIGN_RIGHT);
+                this.jsonInit({
+                  "message0": lf("for %1 from 0 to %2"),
+                  "args0": [
+                    {
+                      "type": "field_variable",
+                      "name": "VAR",
+                      "variable": lf("_default_var") // "item" in English
```

Comment:
Same as ``_for_do`` and ``_while_do``, please open an issue and we'll fix this in a future commit.